assets/js/blocks/products/all-products/edit.tsx

```diff
@@ -48,46 +47,58 @@ import { getSharedContentControls, getSharedListControls } from '../edit';
 import Block from './block';
 import './editor.scss';
 
+type EditorAttributes = {
+	columns: number;
+	rows: number;
+	alignButtons: boolean;
+	contentVisibility: object;
+	orderby: string;
+	layoutConfig: Array< object >;
+	isPreview: boolean;
+};
+
+interface EditorProps {
+	block: BlockInstance;
+	attributes: EditorAttributes;
+	debouncedSpeak: ( label: string ) => void;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+	replaceInnerBlocks: (
+		rootClientId: string,
+		blocks: BlockInstance[],
+		updateSelection?: boolean
+	) => void;
+}
+
+interface EditorState {
+	isEditing: boolean;
+	innerBlocks: BlockInstance[];
+}
+
 /**
  * Component to handle edit mode of "All Products".
  */
-class Editor extends Component {
-	static propTypes = {
-		/**
-		 * The attributes for this block.
-		 */
-		attributes: PropTypes.object.isRequired,
-		/**
-		 * A callback to update attributes.
-		 */
-		setAttributes: PropTypes.func.isRequired,
-		/**
-		 * From withSpokenMessages.
-		 */
-		debouncedSpeak: PropTypes.func.isRequired,
-	};
-
+class Editor extends Component< EditorProps, EditorState > {
```

Comment:
While briefly looking at your changes, I noticed that this file is still using a class component. Now that we're touching this file, we should convert it to a functional component. Could you take care of that?

Reply:
@nielslange I've moved from class to functional component.